### PR TITLE
Move the client, service, schema generation logic to the separate modules

### LINKED ISSUE #1059: Move the client, service, schema generation logic to the separate modules
**Description:**
Might Need to restructure the openAPI tool components in the following way 
```
|-- OpenAPI CLI: contains only the cli command related logic + utilities
|-- Openapi-core: will contain all the generation logic, separated as below
    |---- Client
    |---- Service
    |---- Schema
```

Related issue : https://github.com/ballerina-platform/openapi-tools/issues/1058

### PR DESCRIPTION
**Pull request by lnash94**
Screenshot from 2022-09-05 11-15-20

```
|-- openapi-cli: contains only the cli command related logic + utilities
|-- openapi-core: will contain all the generation logic, separated as below
    |---- client
    |---- document (this module for generating documents in client and schema. service will use this in future once it support to add API doc in generated code)
    |---- service
    |---- schema
```

### CODE REVIEW

File: openapi-cli/src/main/java/io/ballerina/openapi/cmd/OpenApiCmd.java [hunk 1]
```diff
@@ -137,7 +137,7 @@ public void execute() {
         if (inputPath) {
             //Check if an OpenApi definition is provided
             if (argList == null) {
-                outStream.println(OpenApiMesseges.MESSAGE_FOR_MISSING_INPUT);
+                outStream.println(OpenApiMessages.MESSAGE_FOR_MISSING_INPUT);
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/main/java/io/ballerina/openapi/cmd/OpenApiCmd.java:140]:
```suggestion
                outStream.println(OpenAPIMessages.MESSAGE_FOR_MISSING_INPUT);
```

File: openapi-cli/src/main/java/io/ballerina/openapi/cmd/OpenApiMessages.java [hunk 2]
```diff
@@ -15,19 +15,22 @@
...
  * This class contains the messages constants required for OpenApi tool.
+ *
+ * @since 1.3.0
  */
-public class OpenApiMesseges {
+public class OpenApiMessages {
 
     public static final String OPENAPI_CLIENT_EXCEPTION = "Error occurred when generating client for OpenAPI contract";
 
     public static final String MESSAGE_FOR_MISSING_INPUT = "An OpenAPI definition file is required to generate the " +
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/main/java/io/ballerina/openapi/cmd/OpenApiMessages.java:30]:
```suggestion
    public static final String MISSING_CONTRACT_PATH = "An OpenAPI definition path is required to generate the " +
```

File: openapi-cli/src/main/java/io/ballerina/openapi/cmd/OpenApiMessages.java [hunk 3]
```diff
@@ -15,19 +15,22 @@
...
 
 /**
  * This class contains the messages constants required for OpenApi tool.
+ *
+ * @since 1.3.0
  */
-public class OpenApiMesseges {
+public class OpenApiMessages {
 
     public static final String OPENAPI_CLIENT_EXCEPTION = "Error occurred when generating client for OpenAPI contract";
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/main/java/io/ballerina/openapi/cmd/OpenApiMessages.java:28]:
```suggestion
    public static final String CLIENT_GENERATION_FAILED = "Error occurred when generating client for OpenAPI contract";
```

File: openapi-cli/src/main/java/io/ballerina/openapi/cmd/OpenApiMessages.java [hunk 4]
```diff
@@ -15,19 +15,22 @@
...
+
 package io.ballerina.openapi.cmd;
 
 /**
  * This class contains the messages constants required for OpenApi tool.
+ *
+ * @since 1.3.0
  */
-public class OpenApiMesseges {
+public class OpenApiMessages {
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/main/java/io/ballerina/openapi/cmd/OpenApiMessages.java:26]: Shouldn't this be renamed as `ErrorMessages`?

Comment (lnash94) [reply openapi-cli/src/main/java/io/ballerina/openapi/cmd/OpenApiMessages.java:26]: Yeah we can use as `ErrorMessages` will re change it :)

File: openapi-core/src/main/java/io/ballerina/openapi/core/ErrorMessages.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.openapi.core;
+
+/**
+ * Container for error messages of the OpenAPI validator plugin.
```
Comment (NipunaRanasinghe) [inline openapi-core/src/main/java/io/ballerina/openapi/core/ErrorMessages.java:22]: Isn't the validator plugin a separate component outside the openAPI core?

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM. 

As per the offline discussion and the review we had (meeting notes: https://docs.google.com/document/d/1_NkOdgm1wLooAgn6QOzNTQKkighc2UCNddcD3SCt1Wg/edit?usp=sharing), let's work on restructuring the openAPI extensions and the common utils as a separate task.

**Pull request merged**